### DISCUSSION TITLE
allow specifying where to `spawn`

### DISCUSSION THREAD

**Discussion by KiaraGrouwstra** [YaLTeR/niri#248] [Ideas]
we are now able to bind spawn commands to keybinds, trigger them on login, or script them thru `niri msg action`.
however, the current mechanics have new windows spawn in a location depending on the focus workspace and window, while one might imagine wanting to script this otherwise, e.g. to have applications spawned in a specific workspace.
it would be nice if such an option were exposed, if maybe not the highest priority. :)

**Comment by YaLTeR** [comment]
Hmm, I sense a connection to window rules here. I.e. you might want to set window rules like `open-on-output` (or, when it happens, `open-on-workspace`) for a particular spawn (or spawn-at-startup). Need to think how to best do this.

**Reply by KiaraGrouwstra** [reply]
i guess workspace number aside there is also window position within a workspace, e.g. spawn to the left of the active window there, spawn as the right-most window, spawn in position 2 from the left... i dunno what standards exist on this kinda stuff tho, if any.

**Reply by mario-holubar** [reply]
It's been a while since this discussion was started, are there any updates on this?
It seems weird to me to not even have a built-in way to create a new tab in the current column.

**Reply by YaLTeR** [reply]
No update.

> a built-in way to create a new tab in the current column

https://yalter.github.io/niri/FAQ.html#can-i-open-a-window-directly-in-the-current-column-in-the-same-column-as-another-window

**Comment by thiagovianav** [comment]
up on this!
